addons/remix-tree-button/main.js

```diff
@@ -1,35 +1,174 @@
+      document.querySelector("#scratchAddonsRemixTreeBtn").style.display = "block";
+      linkOpensInSameTab();
+    } else {
+      if (addon.tab.editorMode === "projectpage") {
+        addon.tab
+          .waitForElement(".flex-row.subactions", {
+            reduxCondition: (state) => state.scratchGui.mode.isPlayerOnly,
+          })
+          .then(() => {
+            if (!document.querySelector(".copy-link-button")) return;
+            if (addon.self.disabled) return;
+            if (addon.settings.get("type") === "replace") {
+              // Change the link from ".../remixes" to ".../remixtree".
+              addon.tab.waitForElement(".remix-list").then(() => {
+                document.querySelector(".remix-list a").href = window.location.href + "remixtree";
+                if (addon.settings.get("new-tab")) {
+                  linkOpensInNewTab();
+                } else {
+                  linkOpensInSameTab();
+                }
+                document.querySelector(".remix-list a span").textContent = msg("remix-tree-link");
+              });
+            } else {
+              // Add a button next to "Copy Link".
+              linkOpensInSameTab();
+              if (document.querySelector("#scratchAddonsRemixTreeBtn")) return; // Check again because we're inside a promise
+              const remixtree = document.createElement("button");
+              const remixtreeSpan = document.createElement("span");
+              remixtreeSpan.innerText = msg("remix-tree");
+              remixtree.className = "button action-button remixtree-button";
+              remixtree.id = "scratchAddonsRemixTreeBtn";
+              remixtree.appendChild(remixtreeSpan);
+              remixtree.addEventListener("click", (e) => {
+                goToRemixTree(e);
+              });
+              addon.tab.appendToSharedSpace({ space: "afterCopyLinkButton", element: remixtree, order: 0 });
+            }
           });
-          addon.tab.appendToSharedSpace({ space: "afterCopyLinkButton", element: remixtree, order: 0 });
-        });
+      }
+    }
+  }
+
+  function goToRemixTree(e) {
+    // Follow the link to the remix tree after clicking the button.
+    if (addon.settings.get("open-with-icon") && !addon.self.disabled) {
+      if (addon.settings.get("new-tab") || e.ctrlKey || e.metaKey) {
+        window.open(
+          `https://scratch.mit.edu/projects/${window.location.href.split("projects")[1].split("/")[1]}/remixtree`,
+          "_blank",
+          "noopener,noreferrer"
+        );
+      } else {
+        window.location.href = `https://scratch.mit.edu/projects/${
+          window.location.href.split("projects")[1].split("/")[1]
+        }/remixtree`;
+      }
+    }
+  }
+
+  function undoLoadRemixButton() {
+    // Removes the button.
+    if (document.querySelector("#scratchAddonsRemixTreeBtn")) {
+      document.querySelector("#scratchAddonsRemixTreeBtn").style.display = "none";
+    }
+  }
```

Comment:
Could [addon.tab.displayNoneWhileDisabled](https://scratchaddons.com/docs/reference/addon-api/addon.tab/#using-addontabdisplaynonewhiledisabled-dynamicdisable) be used here? It would also mean you wouldn't need to change the display type when the addon is dynamically re-enabled.

Reply:
I'll look into it. It might be more complicated when dealing with settings that can be dynamically changed.

Reply:
Oh, right, I guess that's the problem with having multiple placements for the same addon.